README.md

```diff
@@ -244,6 +252,60 @@ This is useful on development machines. It allows you to:
 
 All while running Vaultenv without any CLI args.
 
+It can happen that conflicting configuration values are send to Vaultenv. An example
+would be a global secret file, which is overwritten by a project specific configuration file.
+The order of precedent (from least important to most important) is as follows:
+ - `/etc/vaultenv.conf`
+ - `$HOME/.config/vaultenv/vaultenv.conf`
+ - `$CWD/.env`
+ - environment variables
+ - command line options
+
+This means that any command line option that is present would overwrite any other configuration.
+If an option is not specified, the default is used. The defaults are as follows:
+```
+Host:           localhost
+Port:           8200
+Addr            https://localhost:8200
+Token:          Unspecified
+Secret file:    Unspecified
+Command:        Unspecified
+Arguments:      []
+Use TLS:        True
+Validate certs: True
+Inherit env:    True
+Base delay:     40
+Retry attempts: 9
+Log-level:      Error
+Use PATH:       True
+```
+In cases where no default nor any value is specified, which is possible for `Token`, `Secret file` and
+`Command`, Vaultenv will give an error that it requires these values to operate.
+
+Vaultenv also supports the `VAULT_ADDR` configuration. In such a case, without specifying separate
```

Comment:
This is starting to become quite a wall of text. 

Can we make this its own section "Connection options"? We could add a general introduction that there are two ways to specify connection options: `VAULT_ADDR` and the specific values. Then the rest of the text is a bit easier to read